papermill/translators.py

```diff
@@ -158,6 +192,84 @@ def codify(cls, parameters, comment='Parameters'):
             content = black.format_str(content, mode=fm)
         return content
 
+    @classmethod
+    def inspect(cls, parameters_cell):
+        """Inspect the parameters cell to get a Parameter list
+
+        It must return an empty list if no parameters are found and
+        it should ignore inspection errors.
+
+        Parameters
+        ----------
+        parameters_cell : NotebookNode
+            Cell tagged _parameters_
+
+        Returns
+        -------
+        List[Parameter]
+            A list of all parameters
+        """
+        params = []
+        src = parameters_cell['source']
+
+        def flatten_accumulator(accumulator):
+            """Flatten a multilines variable definition.
+
+            Remove all comments except on the latest line - will be interpreted as help.
+
+            Args:
+                accumulator (List[str]): Line composing the variable definition
+            Returns:
+                Flatten definition
+            """
+            flat_string = ""
+            for line in accumulator[:-1]:
+                if "#" in line:
+                    comment_pos = line.index("#")
+                    flat_string += line[:comment_pos].strip()
+                else:
+                    flat_string += line.strip()
+            if len(accumulator):
+                flat_string += accumulator[-1].strip()
+            return flat_string
+
+        grouped_variable = []
+        accumulator = []
+        for iline, line in enumerate(src.splitlines()):
```

Comment:
Interesting combination of operations to allow for the regex to pick up the help text. Might be worth a comment explaining why this code is here and why the regex alone isn't sufficient (I had to come back after reading it to get why).

Reply:
I added a comment